pkg/server/status.go

```diff
@@ -4196,3 +4204,60 @@ func (s *statusServer) ListJobProfilerExecutionDetails(
 	}
 	return &serverpb.ListJobProfilerExecutionDetailsResponse{Files: files}, nil
 }
+
+func (s *statusServer) localUpdateTableMetadataCache(
+	ctx context.Context,
+) (*serverpb.UpdateTableMetadataCacheResponse, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	if s.mu.updateTableMetadataJobSignal == nil {
+		return nil, errors.New("no job signal")
+	}
+	success := false
+	select {
+	case s.mu.updateTableMetadataJobSignal <- struct{}{}:
+		success = true
+	default:
+	}
+	return &serverpb.UpdateTableMetadataCacheResponse{Success: success}, nil
+}
+
+func (s *statusServer) UpdateTableMetadataCache(
+	ctx context.Context, req *serverpb.UpdateTableMetadataCacheRequest,
+) (*serverpb.UpdateTableMetadataCacheResponse, error) {
+	ctx = s.AnnotateCtx(ctx)
+	if req.Local {
+		return s.localUpdateTableMetadataCache(ctx)
+	}
+
+	// Get the node id for the job.
+	row, err := s.internalExecutor.QueryRow(ctx, "get-node-id", nil, `
+SELECT claim_instance_id
+FROM system.jobs
+WHERE job_id = $1
+`, jobspb.TypeUpdateTableMetadataCache)
```

Comment:
Small drive-by. I think this should be: `jobs.UpdateTableMetadataCacheJobID`?

Reply:
Depending on your use case, you may want to also want to see if the claim is still valid.

Reply:
🤦‍♀️ Oops, thanks! Changed to the job id. WDYM by still valid?

Reply:
A claim on a job is valid so long as the related sql session is still alive. The job system will _eventually_ clear out claim's for expired sessions, but if you wanted you could check proactively.